### PR TITLE
Support for local: sources

### PR DESCRIPTION
**Pull request by dagss**
Implements https://github.com/hashdist/hashdist/issues/341.

First, it was prudent to change the special github: feature, so instead of doing
```
packages:
    numpy:
        github: http://github.com/numpy/numpy/commits/blah
```
you should do
```
packages:
    numpy:
        sources:
        - github: http://github.com/numpy/numpy/commits/blah
```
i.e., you always replace the entire sources section. Also, the `github:` way of writing a commit can now be used directly in package specs too.

Then, the feature implemented is you can do (in package spec, or overriding sources in profile spec, doesn't matter):

```
sources:
- local: /path/to/my/local/files
```

Basically if the target is not a git repository, everything will be included, hashed, and copied for the build, whereas if it's a git repository then only files under git control will be included. However any modifications to files are included, including modifications to any new files that are added to the index.

TODO:
- [ ] Make sure files added to the index, but not committed, are also included..

### CODE REVIEW

**Comment by dagss** [comment]
PS this depends on dagss/parameters going in first.
